tests/test_views_objectversion.py

```diff
@@ -502,3 +502,30 @@ def test_delete_unwritable(client, db, bucket, versions):
     FileInstance.query.count() == 4
     remove_file_data(obj.file_id)
     FileInstance.query.count() == 4
+
+
+def test_put_header_tags(client, bucket, permissions, get_md5, get_json):
+    """Test upload of an object with tags in the headers."""
+
+    key = 'test.txt'
+    data = b'updated_content'
+    checksum = get_md5(data, prefix=True)
+    object_url = url_for(
+        'invenio_files_rest.object_api', bucket_id=bucket.id, key=key)
+    headers = {
+        'X-Invenio-Tags': 'key1=val1;key2=val2;key3=val3;key1=invalid'
```

Comment:
This actually should fail because you are repeating one of the keys. I would also change the formatting to follow what amazon uses `tag1=value1&tag2=value2`